File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/interfaces/ImgInterface.java
package xyz.jienan.xkcd.whatif.interfaces;

import android.webkit.JavascriptInterface;

public class ImgInterface {

    private ImgCallback imgCallback;

    public ImgInterface(ImgCallback imgCallback) {
        this.imgCallback = imgCallback;
    }

    @JavascriptInterface
    public void doLongPress(String title) {
        if (imgCallback != null) {
            imgCallback.onImgLongClick(title);
        }
    }

    public interface ImgCallback {
        void onImgLongClick(String title);
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/whatif/interfaces/ImgInterface.kt
package xyz.jienan.xkcd.whatif.interfaces

import android.webkit.JavascriptInterface

class ImgInterface(private val imgCallback: ImgCallback?) {

    @JavascriptInterface
    fun doLongPress(title: String) {
        imgCallback?.onImgLongClick(title)
    }

    interface ImgCallback {
        fun onImgLongClick(title: String)
    }
}
